server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconcilerMetricsIT.java

```diff
@@ -73,7 +84,7 @@ public void testDesiredBalanceMetrics() {
         );
         assertThat(desiredBalanceNodeWeightsMetrics.size(), equalTo(2));
         for (var nodeStat : desiredBalanceNodeWeightsMetrics) {
-            assertThat(nodeStat.value().doubleValue(), greaterThanOrEqualTo(0.0));
+            assertTrue(nodeStat.isDouble());
```

Comment:
Correct, this might happen if all components are below average in
https://github.com/elastic/elasticsearch/blob/69150c8de16335e0c3444d6bf5ccb4b705d359ad/server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java#L303-L305

For example if we start an empty node I suspect it would have the negative weight.
I hope this could be displayed correctly on the chart.